congruity/congruity.py

```diff
@@ -1863,8 +1863,14 @@ def main():
             else:
                 raise CmdLineException("ERROR: Option '%s' not recognized" % arg)
         if len(argv) != 1:
-            raise CmdLineException("ERROR: Precisely one filename argument is required")
-        ezhex_filename = argv.pop(0)
+            # We did not get a file name on the command line, prompt the user for one.
+            with wx.FileDialog(None, "Congruity - Open a file", wildcard="EZHex files (*.ez*)|*.ez*",
+                               style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST) as fileDialog:
+                if fileDialog.ShowModal() == wx.ID_CANCEL:
+                    os._exit(1)
```

Comment:
Why `os._exit()` and not `sys.exit()`?  Also, why a non-zero code?

Reply:
Good questions. I'd certainly prefer to terminate as cleanly as possible. Unfortunately, Python crashes if I use `sys.exit()` or `return`. Additionally, the exception handler in `main()` tries to catch the exception `sys.exit` uses to exit and so I have to specifically pass it through if I don't want Congruity to throw an extra error message. Lastly, from before my time in this codebase, I see a whole bunch of usage of `os._exit` in `congruity.py` and `mhgui.py` and no instances of `sys.exit` .

As for the exit code: Whenever I have written a program that requires a file as input, I have always interpreted "user does not provide the file as requested and thus the program cannot run" as an abnormal termination and hence the non-zero exit code. (Most valuable in cases where you're running a program from a batch file or shell script and thus the rest of the script can't proceed because the prerequisite program didn't do its job.) But do whatever you think is best.

Reply:
OK, fair enough.  I think the use of `os._exit()` predates even my time on this project, so we can leave it.